oximeter/collector/src/agent.rs

```diff
@@ -387,7 +388,7 @@ impl OximeterAgent {
         address: SocketAddrV6,
         refresh_interval: Duration,
         db_config: DbConfig,
-        resolver: &Resolver,
+        resolver: BoxedResolver,
```

Comment:
In this PR, basically anywhere we pass a DNS resolver, I instead try to pass around a [qorb::pool::Pool](https://docs.rs/qorb/0.0.1/qorb/pool/struct.Pool.html), or a boxed [DnsResolver](https://docs.rs/qorb/0.0.1/qorb/resolvers/dns/struct.DnsResolver.html), which is an argument used to construct a Pool.

This pool does a couple things:

1. It performs DNS resolution in the background, to find a set of available backends
2. It automatically calls the [connect](https://docs.rs/qorb/0.0.1/qorb/backend/trait.Connector.html#tymethod.connect) method -- for whatever connector implementation you use.
3. It lets you re-use connections as they are returned into the pool.

As backends (Nexus, Clickhouse, etc, whatever service you're pointing at) come and go, the pool should automatically adjust the set of connections to each that are available.